### PR TITLE
[Product Block Editor]: expose __experimentalSectionActions component

### PR DESCRIPTION
**Pull request by retrofox**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

This PR exposes the `__experimentalSectionActions` component to be used externally. Also, it adds a basic Readme.md file.

Closes # .

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

* Enable the new product editor (beta) in the Settings > Advanced > Features page
* Add/edit a product
* Ensure the app compiles properly 
* Ensure the blocks that use these components keep working as usual. For instance:

Screenshot 2024-04-12 at 09 26 40

* Also, a visual review should be more than enough, but if you want to check if the component is being exposed:

1. Open the nextwork tab of the dev tools of your browser
2. Filter by `JS` files
3. Filter by the `product-editor/index` name
4. You should see only one js file listed
image

6. Open it in a new tab (usually, double click works)

7. Search the `__experimentalSectionActions` component
8. Confirm it's defined in the build code

image

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [x] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [x] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: packages/js/product-editor/src/components/block-slot-fill/section-actions/Readme.md [hunk 1]
```diff
@@ -0,0 +1,28 @@
+# SectionActions
+
+`<SectionActions />` is a React component designed to be used within blocks in WooCommerce Product Editor,
+providing a slot for specific actions related to the section in which it is included.
+
+## Example
+
+```jsx
+import { Button } from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
```
Comment (louwie17) [inline packages/js/product-editor/src/components/block-slot-fill/section-actions/Readme.md:11]: Should we also add the `SectionActions` import here as part of the example?
```
import { __experimentalSectionActions as SectionActions } from '@woocommerce/product-editor'
```

Comment (retrofox) [reply packages/js/product-editor/src/components/block-slot-fill/section-actions/Readme.md:11]: addressed

**Review by jimjasson — APPROVED** [review]
Tested this locally with a custom subsection and I was able to add custom actions using the `SectionActions` component as expected.

**Review by louwie17 — APPROVED** [review]
LGTM 🚀

**Pull request merged**